efel/pyfeatures/pyfeatures.py

```diff
@@ -57,6 +59,41 @@ def time():
     return _get_cpp_feature("time")
 
 
+def impedance():
+    from scipy.ndimage.filters import gaussian_filter1d
+
+    dt = _get_cpp_data("interp_step")
+    Z_max_freq = _get_cpp_data("impedance_max_freq")
+    voltage_trace = voltage()
+    holding_voltage = _get_cpp_feature("voltage_base")
+    normalized_voltage = voltage_trace - holding_voltage
+    current_trace = current()
+    if current_trace is not None:
+        holding_current = _get_cpp_feature("current_base")
+        normalized_current = current_trace - holding_current
+        spike_count = _get_cpp_feature("Spikecount")
+        if spike_count < 1:  # if there is no spikes in ZAP
+            fft_volt = numpy.fft.fft(normalized_voltage)
+            fft_cur = numpy.fft.fft(normalized_current)
+            # convert dt from ms to s to have freq in Hz
+            freq = numpy.fft.fftfreq(len(normalized_voltage), d=dt / 1000.)
+            if any(fft_cur) == 0:
+                return None
```

Comment:
Here also, to check this condition we don't need to evaluate the `numpy.fft.fftfreq`. In fact the else branch below is also can be removed since the condition above is a guard condition that returns.

```python
  ...
  fft_volt = numpy.fft.fft(normalized_voltage)
  fft_cur = numpy.fft.fft(normalized_current)
  if any(fft_cur) == 0:
      return None
  # convert dt from ms to s to have freq in Hz
  freq = numpy.fft.fftfreq(len(normalized_voltage), d=dt / 1000.)
  Z = fft_volt / fft_cur
  norm_Z = abs(Z) / max(abs(Z))
  select_idxs = numpy.swapaxes(
  numpy.argwhere((freq > 0) & (freq <= Z_max_freq)), 0, 1
  )[0]
  smooth_Z = gaussian_filter1d(norm_Z[select_idxs], 10)
  ind_max = numpy.argmax(smooth_Z)
  ...
```

Reply:
Or 

```python
...
fft_volt = numpy.fft.fft(normalized_voltage)
fft_cur = numpy.fft.fft(normalized_current)
if any(fft_cur) == 0:
	return None
else:
	# convert dt from ms to s to have freq in Hz
	freq = numpy.fft.fftfreq(len(normalized_voltage), d=dt / 1000.)
	Z = fft_volt / fft_cur
	norm_Z = abs(Z) / max(abs(Z))
	select_idxs = numpy.swapaxes(
	numpy.argwhere((freq > 0) & (freq <= Z_max_freq)), 0, 1
	)[0]
	smooth_Z = gaussian_filter1d(norm_Z[select_idxs], 10)
	ind_max = numpy.argmax(smooth_Z)
...
```